test/Microsoft.AspNetCore.Mvc.Core.Test/RedirectToActionResultTest.cs

```diff
@@ -83,6 +83,41 @@ public class RedirectToActionResultTest
                 "No route matches the supplied values.");
         }
 
+        [Fact]
+        public async Task RedirectToAction_Execute_WithFragment_PassesCorrectValuesToRedirect()
```

Comment:
Add a similar test in `RedirectToRouteResultTest`, perhaps covering the overload accepting `permanent: true` instead.